presto-main/src/main/java/com/facebook/presto/execution/Lifespan.java

```diff
@@ -67,7 +71,22 @@ public static Lifespan jsonCreator(String value)
         return Lifespan.driverGroup(parseInt(value.substring("Group".length())));
     }
 
+    @ThriftConstructor
```

Comment:
While we're doing this exercise, it would be nice not have to do String parsing for these values.  Did you consider an alternate constructor, perhaps that takes in a boolean with OptionalInt, and validates that when it's not grouped it's empty, and when it's grouped it's not empty?

Reply:
This logic was sort of mimicking what the json factory method was doing.
`public static Lifespan jsonCreator(String value)`

Reply:
I wonder if we can use jsonCreator(String value) here.  We can rename the method. createLifeSpan?

Reply:
@cemcayiroglu  don't think a constructor can refer to a factory method directly. The alternate is going the builder route which in drift requires a complete Builder class. We can try adding in support for factory methods similar to what Jackson does as a good enhancement later on.